paddle/c/scope.cc

```diff
@@ -0,0 +1,32 @@
+#include "scope.h"
```

Comment:
If we use Boost::Python, all the following files in this PR

- scope.h 
- scope.cc
- private.h
- error.h
- dllexports.h

would be replaced by the following few-line C++ code:

```cpp
#include <boost/python.hpp>

BOOST_PYTHON_MODULE(pypaddle) {
    boost::python::class_<paddle::framework::Scope>("Scope")
        .def("create_var", &Scope::CreateVar)
        .def("create_scope", &Scope::CreateScope);
}
```

Please re-think about this necessity of create a C-API.